src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -317,12 +317,27 @@ def purge(self, service_name: str) -> None:
             '--namespace', cast(str, spec.service_id),
             'rm', 'grace',
         ]
-        subprocess.run(
-            cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            timeout=10
-        )
+        try:
+            result = subprocess.run(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                timeout=10
+            )
+        except Exception as ex:
```

Comment:
It's better to catch the narrowest exception possible, so `SubprocessError` otherwise you could hide other pogramming errors (say if you typed subprocess.PIP it would supress an attribute error).